extensions/HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/VMGalleryVMCreationOperation.cs

```diff
@@ -85,67 +89,82 @@ public void Report(IOperationReport value)
 
-                await DownloadImageAsync();
-                var virtualMachineHost = _hyperVManager.GetVirtualMachineHost();
-                var absoluteFilePathForVhd = GetUniqueAbsoluteFilePath(virtualMachineHost.VirtualHardDiskPath);
+                    // extract the archive file to the destination file.
+                    var archiveProvider = _archiveProviderFactory.CreateArchiveProvider(ArchivedFile!.FileType);
 
-                // extract the archive file to the destination file.
-                var archiveProvider = _archiveProviderFactory.CreateArchiveProvider(ArchivedFile!.FileType);
+                    UpdateProgress(_stringResource.GetLocalized("ExtractionStarting", $"({Image.Name})"));
+                    await archiveProvider.ExtractArchiveAsync(this, ArchivedFile!, absoluteFilePathForVhd, CancellationTokenSource.Token);
+                    var virtualMachineName = MakeFileNameValid(_userInputParameters.NewEnvironmentName);
 
-                await archiveProvider.ExtractArchiveAsync(this, ArchivedFile!, absoluteFilePathForVhd, CancellationTokenSource.Token);
-                var virtualMachineName = MakeFileNameValid(_userInputParameters.NewEnvironmentName);
+                    // Use the Hyper-V manager to create the VM.
+                    UpdateProgress(_stringResource.GetLocalized("CreationInProgress", virtualMachineName));
+                    var creationParameters = new VirtualMachineCreationParameters(
+                        _userInputParameters.NewEnvironmentName,
+                        GetVirtualMachineProcessorCount(),
+                        absoluteFilePathForVhd,
+                        Image.Config.SecureBoot,
+                        Image.Config.EnhancedSessionTransportType);
 
-                // Use the Hyper-V manager to create the VM.
-                UpdateProgress(_stringResource.GetLocalized("CreationInProgress", virtualMachineName));
-                var creationParameters = new VirtualMachineCreationParameters(
-                    _userInputParameters.NewEnvironmentName,
-                    GetVirtualMachineProcessorCount(),
-                    absoluteFilePathForVhd,
-                    Image.Config.SecureBoot,
-                    Image.Config.EnhancedSessionTransportType);
+                    return new CreateComputeSystemResult(_hyperVManager.CreateVirtualMachineFromGallery(creationParameters));
+                }
+                catch (Exception ex)
+                {
+                    if (creationAttempt == maxAttempts)
+                    {
+                        _log.Error(ex, "Operation to create compute system failed on the last attempt");
+                        return new CreateComputeSystemResult(ex, ex.Message, ex.Message);
+                    }
+                    else
+                    {
+                        _log.Error(ex, "Operation to create compute system failed on the first attempt, retrying.");
```

Comment:
This message makes no sense if max attempts == 3 and failed on the 2nd attempt.   Remove references to the "first attempt"

Reply:
updated thanks